spanner/cloudexecutor/executor/cloud_executor_util.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+// cloud_executor_util.go file contains utility methods that can be reused across executor actions.
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"cloud.google.com/go/spanner"
+	"cloud.google.com/go/spanner/apiv1/spannerpb"
+	executorpb "cloud.google.com/go/spanner/cloudexecutor/proto"
+	spb "google.golang.org/genproto/googleapis/rpc/status"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	_ "google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// tableMetadataHelper is used to hold and retrieve metadata of tables and columns involved
+// in a transaction.
+type tableMetadataHelper struct {
```

Comment:
Suggestion - Would be better to carve this to a new file. `tableMetadataHelper` looks like a good enough independent responsibility in itself.